doc/sr_integration.rst

```diff
@@ -0,0 +1,211 @@
+
+    #include "client.h"
+    void* client = NULL;
+    return_code = SmartRedisCClient(use_cluster), &client
+
+All these methods only have one configurable parameter, indicated in the
+above cases by the variable `use_cluster`. If this parameter, is true,
+then the client expects to be able to communicate with a database with
+three or more shards.
+
+++++++++++++++++++++++++++++++++++
+(Parallel Programs): Creating keys
+++++++++++++++++++++++++++++++++++
+
+For parallel applications, each rank or thread that is communicating with
+the database will likely need to create a unique prefix for keys to prevent
+another rank or thread inadvertently overwriting data. Any identifier can be
+used, though typically the MPI rank number (or equivalent identifier) is a
+useful, unique number.
+
+C++::
+
+    const std::string key_prefix << std::format("{:06}_", *rank_id);
+
+Python::
+
+    key_prefix = f"{rank_id:06d}_"
+
+Fortran::
+
+    character(len=12) :: key_prefix
+    write(key_prefix,'(A,I6.6)') rank_id
+
+C::
+
+    char[7] key_prefix;
+    key_prefix = sprintf(key_prefix", "%06d" rank_id);
+
+++++++++++++++++++++++++++
+Storing scripts and models
+++++++++++++++++++++++++++
+
+The last task that typically needs to be done is to store models or scripts
+that will be used later in the simulation. When using a clustered database,
+this only needs to be done by one client (unless ranks required a specific
+model or script).
+
+C++::
+
+    if (root_client) {
+        client.set_model_from_file(model_key, model_file, backend, device)
+    }
+
+Python::
+
+    if root_client:
+        client.set_model_from_file(model_key, model_file, backend, device)
+
+Fortran::
+
+    if (root_client) return_code = client%set_model_from_file(model_key, model_file, "TORCH", "CPU")
+
+C::
+
+    if (root_client) {
+        return_code = client.set_model_from_file(client, model_key, model_file, backend, device)
+    }
+
+=========
+Main loop
+=========
+
+Within the main loop of the code (e.g. every timestep or iteration of a solver),
+the developer uses the SmartRedis client methods to implement a workflow which
+may include receiving data, sending data, running a script or model, and/or
+retrieving a result. These workflows are covered extensively in the walkthroughs
+for the Fortran, C++, and python clients and the integrations with MOM6, OpenFOAM,
+LAMMPS, and others.
+
+Generally though, developers are advised to
+
+1. Find locations where file I/O would normally happen and either replace
+   or add code to use the SmartRedis client and store the data in the
+   database
+2. Use the `key_prefix` created during initialization to avoid accidental
+   writes/reads from different clients
+3. Use the SmartSim `dataset` type when using clients representing decomposed
+   subdomains to make the retrieval/use of the data more performant
+
+============
+Full example
+============
+
+The following pseudocode is used to demonstrate various aspects of instrumenting an
```

Comment:
Would the typical reader be more comfortable with C++ than Fortran?

Reply:
We've been fielding more queries from the Fortran dev community than the C++ one. We can always include more examples later on down the line, but for a first pass I wanted to keep it in one language.